core/CMakeLists.txt

```diff
@@ -76,6 +81,15 @@ if (GINKGO_HAVE_PAPI_SDE)
     list(APPEND GKO_RPATH_ADDITIONS "${GKO_PAPI_LIBDIR}")
 endif()
 
+if (GINKGO_BUILD_MPI)
+    target_link_libraries(ginkgo PUBLIC MPI::MPI_CXX)
+    target_include_directories(ginkgo SYSTEM PUBLIC ${MPI_INCLUDE_PATH})
+
+    # Propagate some useful information
+    set(MPI_C_VERSION ${MPI_C_VERSION} PARENT_SCOPE)
+    set(MPI_C_LIBRARIES ${MPI_C_LIBRARIES} PARENT_SCOPE)
```

Comment:
Why do we need this? Setting variables in the parent scope feels like bad practice.